### PR TITLE
fix: appointment slots start and end time

### PR DESCRIPTION
**Pull request by SebastianKrupinski**
Resolves: #6407 
Resolves: #5913

### CODE REVIEW

File: tests/php/unit/Controller/BookingControllerTest.php [hunk 1]
```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
...
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezonePlus10(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Australia/Sydney';
+		$sDT = (new DateTime('2024-6-30 14:00:00'))->getTimestamp();
+		;
```
Comment (st3iny) [inline tests/php/unit/Controller/BookingControllerTest.php]:
```suggestion
```

File: tests/php/unit/Controller/BookingControllerTest.php [hunk 2]
```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
...
+		$this->bookingService->expects(self::once())
+			->method('getAvailableSlots')
+			->with($apptConfg, $sDT, $eDT);
 
-		$this->controller->getBookableSlots($apptConfg->getId(), $start, 'Europe/Berlin');
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezoneMinus10(): void {
+
```
Comment (ChristophWurst) [inline tests/php/unit/Controller/BookingControllerTest.php:197]: nit: empty first line of method body is unusual

File: tests/php/unit/Controller/BookingControllerTest.php [hunk 3]
```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
...
+	public function testGetBookableSlotsTimezonePlus10(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Australia/Sydney';
+		$sDT = (new DateTime('2024-6-30 14:00:00'))->getTimestamp();
+		;
+		$eDT = (new DateTime('2024-7-1 14:00:00'))->getTimestamp();
+		;
```
Comment (st3iny) [inline tests/php/unit/Controller/BookingControllerTest.php]:
```suggestion
```

**Review by ChristophWurst — APPROVED** [review]

**Review by GretaD — APPROVED** [review]

File: tests/php/unit/Controller/BookingControllerTest.php [hunk 4]
```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
...
+		$this->controller->getBookableSlots($apptConfg->getId(), $start, $timezone);
+	}
+
+	public function testGetBookableSlotsTimezonePlus14(): void {
+
+		$now = (new DateTime('2024-6-30 8:00:00'))->getTimestamp();
+		$start = (new DateTime('2024-7-1 4:00:00'))->getTimestamp(); // Start date with America/Toronto offset
+		$timezone = 'Pacific/Kiritimati';
+		$sDT = (new DateTime('2024-6-30 10:00:00'))->getTimestamp();
+		;
```
Comment (st3iny) [inline tests/php/unit/Controller/BookingControllerTest.php]:
```suggestion
```

File: tests/php/unit/Controller/BookingControllerTest.php [hunk 5]
```diff
@@ -151,23 +167,108 @@ public function testGetBookableSlotsInvalidTimezone(): void {
...
+		$eDT = (new DateTime('2024-7-1 10:00:00'))->getTimestamp();
+		;
```
Comment (st3iny) [inline tests/php/unit/Controller/BookingControllerTest.php]:
```suggestion
```

File: tests/php/unit/Controller/BookingControllerTest.php (hunk 2 above)
Comment (SebastianKrupinski) [reply tests/php/unit/Controller/BookingControllerTest.php]: Corrected

**Review by st3iny — APPROVED** [review]
Tested and works. Time zones are tricky ...

File: tests/php/unit/Controller/BookingControllerTest.php (hunk 5 above)
Comment (SebastianKrupinski) [reply tests/php/unit/Controller/BookingControllerTest.php]: Yeah not sure how those got in there. Corrected.

**Pull request merged**